wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java

```diff
@@ -93,7 +94,8 @@ public OAuthToken getToken(String userId) throws IOException {
       if (token == null
           || token.getToken() == null
           || token.getToken().isEmpty()
-          || getJson(gitlabUserEndpoint, token.getToken(), GitLabUser.class) == null) {
+          || isNullOrEmpty(
```

Comment:
@vinokurig could you clarify the principle difference and how it fixes gitlab validation? can we add a unit test for that ?

Reply:
> could you clarify the principle difference and how it fixes gitlab validation?

Currently, in the `main` branch  we check that the `GitLabUser` object is not `null`. If we request the GitLab user with an invalid token, we will not get `null`, but a `GitLabUser` object with empty fields. To fix that I changed the check to verify if the `userId` field is not `null`.
> can we add a unit test for that ?

Done